components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLDatasetProcessor.java

```diff
@@ -351,4 +351,20 @@ public void deleteDatasetVersion(int tenantId, String userName, long versionsetI
         }
     }
 
+    public List<String> getFeatureNames(long datasetId, String featureType) throws MLDataProcessingException {
+        try {
+            return databaseService.getFeatureNames(datasetId, featureType);
+        } catch (DatabaseHandlerException e) {
+            throw new MLDataProcessingException(e.getMessage(), e);
+        }
+    }
+
+    public String getSummaryStats(long datasetId, String featureName) throws MLDataProcessingException {
+        try {
+            return databaseService.getSummaryStats(datasetId, featureName);
+        } catch (DatabaseHandlerException e) {
+            throw new MLDataProcessingException(e.getMessage(), e);
```

Comment:
I think here we are wrapping the message of DatabaseHandlerException from previous layer. Please check other methods.